builders/build/substrate-api/chopsticks.md

```diff
@@ -0,0 +1,167 @@
+
+### Interacting with a Fork {: #interacting-with-a-fork }
+
+When running a fork, by default it will be accessible at `ws://localhost:8000`. You will be able to interact with the parachain via libraries such as [polkadot.js](https://github.com/polkadot-js/common){target=_blank} and its [locally built user interface](https://github.com/polkadot-js/apps){target=_blank}.  
+
+The Polkadot JS Apps user interface can be built with Docker by running the following command:  
+
+```
+docker run --rm -it --name polkadot-ui -e WS_URL=ws://someip:9944 -p 80:80 jacogr/polkadot-js-apps:latest
+```
+
+Afterwards, you should be able to go to `localhost` in the browser to see the UI. To interact with the local fork via the UI:
+
+1. Click the icon in the top left
+2. Go to the bottom and open **Development**
+3. Select the **Custom** endpoint and enter `ws://localhost:8000`
+4. Click the **Switch** button
+
+![Open WSS](/images/builders/build/substrate-api/chopsticks/chopsticks-1.png)
+![Switch WSS](/images/builders/build/substrate-api/chopsticks/chopsticks-2.png)
+
+You should now be able to interact with the fork as you would in the hosted version of Polkadot JS apps.
+
+## Replaying Blocks {: #replaying-blocks }
+
+In the case where you would like to replay a block and retrieve its information to dissect the effects of an extrinsic, you can use the `yarn start run-block` command. Its following flags are:  
+
+|           Flag           |                                           Description                                               |
+|:------------------------:|:---------------------------------------------------------------------------------------------------:|
+|         endpoint         | The endpoint of the parachain to fork.                                                              |
+|          block           | Use to specify at which block hash or number to replay the fork.                                    |
+|      wasm-override       | Path of the WASM to use as the parachain, instead of forking an endpoint.                           |
+|            db            | Path to the name of the file that stores or will store the parachain's database.                    |
+|          config          | Path to the config file.                                                                            |
+| output-path=/[file_path] | Use to print out results to a JSON file instead of printing it out in the console.                  |
+|           html           | Include to generate an HTML representation of the storage diff preview between blocks.              |
+|           open           | Whether to open the HTML representation.                                                            |
+
+For example, running the following command will re-run Moonbeam's block 1000, and write the storage diff and other data in a `moonbeam-output.json` file:  
```

Comment:
can you change what happens in block 1000 to simulate changes to storage? or is the sole purpose of this just to see how the storage changes from one block to another given the preexisting extrinsics?

Reply:
You would be able to use this in the dry-run command for Chopsticks, but it currently doesn't work for Moonbeam networks due to some issue with a state trie

https://github.com/AcalaNetwork/chopsticks/issues/203